test/unit/python/sample_files/write_init_files/physics_inputs_constituent_dim.F90

```diff
@@ -0,0 +1,404 @@
+            call ccpp_physics_suite_variables(suite_names(suite_idx), ccpp_required_data, errmsg, errflg, input_vars=.true., output_vars=.false.)
+
+         ! Loop over all required variables and read from file if uninitialized:
+         do req_idx = 1, size(ccpp_required_data, 1)
+
+            ! Find IC file input name array index for required variable:
+            name_idx = find_input_name_idx(ccpp_required_data(req_idx), use_init_variables, constituent_idx)
+
+            ! Check for special index values:
+            select case (name_idx)
+
+               case (init_mark_idx)
+
+                  ! If variable is already initialized, then do nothing.
+
+               case (no_exist_idx)
+
+                  ! If an index was never found, then save variable name and check the rest of the variables, after which the model simulation will
+                  ! end:
+                     missing_required_vars(len_trim(missing_required_vars)+1:) = trim(sep)//trim(ccpp_required_data(req_idx))
+
+                  ! Update character separator to now include comma:
+                  sep = ', '
+
+               case (prot_no_init_idx)
+
+                  ! If an index was found for a protected variable, but that variable was never marked as initialized, then save the variable name
+                  ! and check the rest of the variables, after which the model simulation will end:
+                     protected_non_init_vars(len_trim(protected_non_init_vars)+1:) = trim(sep2)//trim(ccpp_required_data(req_idx))
+
+                  ! Update character separator to now include comma:
+                  sep2 = ', '
+
+               case (const_idx)
+
+                  ! If an index was found in the constituent hash table, then do nothing, this will be handled later
+
+               case default
+
+                  ! Read variable from IC file:
+
+                  select case (trim(phys_var_stdnames(name_idx)))
+                     case ('potential_temperature')
+                        call read_field(file, 'potential_temperature', input_var_names(:,name_idx), 'lev', timestep, theta)
+
+                     case ('air_pressure_at_sea_level')
+                        call read_field(file, 'air_pressure_at_sea_level', input_var_names(:,name_idx), timestep, slp)
+
+                     case ('super_cool_cat_every_const')
+                        call read_constituent_dimensioned_field(const_props, file, 'super_cool_cat_every_const', input_var_names(:,name_idx),          &
+                             timestep, cool_cat_for_each_const)
+
+                     case ('super_cool_cat_with_default_every_const')
+                        call read_constituent_dimensioned_field(const_props, file, 'super_cool_cat_with_default_every_const',                          &
+                             input_var_names(:,name_idx), timestep, cool_default_cat_for_each_const)
```

Comment:
wouldn't we expect this bit of generated code to be passing in `error_on_not_found=.false.` because there's an initial value set?